File: DrinkMachine.py
from VendMachine import VendMachine
from Drink import Drink


class DrinkMachine(VendMachine):

    def __init__(self):
        super().__init__()
        self.__allProducts = list()

    def setProduct(self, product):
        if isinstance(product, Drink):
            self.__allProducts.append(product)

    def allProduct(self):
        for drink in self.__allProducts:
            print(drink)

File: DrinkMachine.java
package lesson_4;

import java.util.ArrayList;
import java.util.List;

public class DrinkMachine<T> implements VendMachine {
    private T drinkMachine;
    private List<T> finishProduct = new ArrayList<>();

    public T getDrinkMachine() {
        return drinkMachine;
    }

    public void setDrinkMachine(T drinkMachine) {
        this.drinkMachine = drinkMachine;
    }

    public void finishProduct() {
        System.out.println(finishProduct);
    }

    public void setProduct(T drinkMachine) {
        finishProduct.add(drinkMachine);
    }
}